src/NServiceBus.Gateway/Configuration/IGatewayDeduplicationConfiguration.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace NServiceBus.Gateway
+{
+    using ObjectBuilder;
+
+    /// <summary>
+    /// Configures the deduplication storage.
+    /// </summary>
+    public interface IGatewayDeduplicationConfiguration
```

Comment:
Is there a benefit of this being an `interface`? Thinking that if this was an `abstract class` it would be easier to evolve WO breaking changes?

Reply:
no specific reason tbh. I was thinking about an abstract class as well but couldn't find a clear advantage. Maybe the reason you stated might be good enough, happy to change it as I don't prefer a specific approach and the abstract class is probably closer to the core's API.

Reply:
👍 for me being able to evolve without breaking SemVer is nice advantage

Reply:
changed the configuration API to an abstract class, have another look.